bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/view/submit/SubmitView.scala

```diff
@@ -135,18 +133,20 @@ class SubmitView(ph: ProblemRobot, newShellHandler: (Model,Option[File]) => Unit
           for (m <- panel.model) {
             if (saveHandler()) {
               GlassLabel.show(panel.peer.getRootPane, "Submitting Proposal...")
-              submitClient.submit(m.proposal) {psr =>
+              submitClient.submit(m.proposal) { psr =>
                 SwingUtilities.invokeLater(new Runnable {
                   def run() {
-                    dsrs = psr.results.map(a => (a.destination, a.result)).toMap
-                    val m0 = Model.proposal.set(m, psr.proposal)
-                    panel.model = Some(m0)
-                    saveHandler()
                     GlassLabel.hide(panel.peer.getRootPane)
 
+                    dsrs = psr.results.map(a => (a.destination, a.result)).toMap
                     val errors = psr.results.map(_.result).filter(!_.isSuccess)
-                    // Show prompt with error messages
-                    if (errors.nonEmpty) {
+
+                    if (errors.size != psr.results.size) {
+                      val m0 = Model.proposal.set(m, psr.proposal)
+                      panel.model = Some(m0)
+                      saveHandler()
+                    } else if (errors.nonEmpty) {
+                      // Show prompt with error messages
```

Comment:
Also, I think the intent might be clearer with:

````scala
  if (psr.results.exists(_.result.isSuccess)) {
    ...
    saveHandler()
  }
````

instead of comparing sizes?